apps/src/netsim/NetSimRouterNode.js

```diff
@@ -1103,11 +1103,20 @@ NetSimRouterNode.prototype.forwardMessageToRecipient_ = function (message, onCom
 
   // TODO: Handle bad state where more than one wire matches dest address?
 
+  // Normally the recipient simulates a message.
+  // If this message is on loopback (i.e. to or from auto-dns) then
+  // the simulator of the original message has to simulate this one too.
+  var simulatingNode = destinationNodeID;
+  if (destinationNodeID === this.entityID) {
+    simulatingNode = message.simulatedBy;
+  }
+
   // Create a new message with a new payload.
   NetSimMessage.send(
       this.shard_,
       routerNodeID,
       destinationNodeID,
+      simulatingNode,
```

Comment:
Picking a simulating node, #1:
When forwarding a message, the router always sets the recipient as the simulating node, UNLESS the message is looping back to the router; then the original simulating node is preserved.